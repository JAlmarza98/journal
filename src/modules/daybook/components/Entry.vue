<template> 
    <div class="entry-contariner mb-3 pointer p-2" @click="$router.push({name: 'entry', params: {id:entry.id}})">
        <div class="entry-title d-flex">
            <span class="text-success fs-5 fw-bold">{{ day }}</span>
            <span class="mx-1 fs-5">{{ month }}</span>
            <span class="mx-2 fw-light">{{ yearDay }}</span>
        </div>

        <div class="entry-description">
            {{ formatText }}
        </div>
    </div>
</template>

<script>
import { months, days } from '../helpers/getDayMonthYear'

export default {
    props: {
        entry: {
            type: Object,
            required:true
        }
    },
    computed: {
        formatText() {
            return(this.entry.text.length > 140)
                ? this.entry.text.substring(0,137) + '...'
                : this.entry.text
        },
        day() {
            const date = new Date( this.entry.date )
            return date.getDate()
        }, 
        month() {
            const date = new Date( this.entry.date )
            return months[date.getMonth()]
        },
        yearDay() {
            const date = new Date( this.entry.date )
            return `${ date.getFullYear() }, ${ days[ date.getDay() ] }`
        }
    }
}
</script>

<style lang="scss" scoped>
.entry-contariner {
    border-bottom: 1px solid #2c3e50;
    transition: 0.2s all ease-in;

    &:hover {
        background-color: lighten($color: grey, $amount: 45);
        transition: 0.2s all ease-in;
    }
}

.entry-description {
    font-size: 12px;
}

</style>