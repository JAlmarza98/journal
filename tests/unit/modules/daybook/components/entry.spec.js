import { shallowMount } from "@vue/test-utils";
import Entry from '@/modules/daybook/components/Entry.vue'
import { journalState } from "../../../mock/test-journal-state"



describe('Pruebas en el Entry Component', () => {

    const mockRouter = {
        push: jest.fn()
    }

    const wrapper = shallowMount( Entry, {
        props: {
            entry:journalState.entries[0]
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }
    } )

    test('Debe hacer match con el snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    });

    test('Debe reaccionar al hacer click en el entry-container', () => {
        const entryContainer = wrapper.find('.entry-contariner ')
        entryContainer.trigger('click')
        
        expect( mockRouter.push ).toHaveBeenCalledWith(
            {
                name: 'entry', 
                params: {
                    id: journalState.entries[0].id
                }
            }
        )
    });

    test('Pruebas en las propiedades computadas', () => {

        expect( wrapper.vm.day ).toBe(3)
        expect( wrapper.vm.month ).toBe('Marzo')
        expect( wrapper.vm.yearDay).toBe('2022, Jueves')

    });

});