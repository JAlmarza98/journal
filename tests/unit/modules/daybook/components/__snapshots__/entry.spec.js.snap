// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Entry Component Debe hacer match con el snapshot 1`] = `
<div class="entry-contariner mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">3</span><span class="mx-1 fs-5">Marzo</span><span class="mx-2 fw-light">2022, Jueves</span></div>
  <div class="entry-description">Estoy aprendiendo Vue.js a buen ritmo</div>
</div>
`;
